### PR TITLE
Adds Windows build information as a label on the node

### PR DESCRIPTION
**Pull request by gab-satchi**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
Kubelet adds a label `node.kubernetes.io/windows-build` on Windows nodes that specifies the Windows Server build information of the node in the following format: `MajorVersion.MinorVersion.BuildNumber`. An example would be: `10.0.17763`

**Special notes for your reviewer**:
While testing the functionality, I discovered that updating an existing node does not include the new label. Deleting the node and recreating does bring in the label. I'm unsure if this is intentional behaviour with how nodes are updated vs. created. 

**Does this PR introduce a user-facing change?**:
```release-note
Adds Windows Server build information as a label on the node.
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:
```docs
- [KEP]: https://github.com/kubernetes/enhancements/pull/1302
```

### CODE REVIEW

**Review by mattjmcnaughton — COMMENTED** [review]
Can you share more about `While testing the functionality, I discovered that updating an existing node does not include the new label. Deleting the node and recreating does bring in the label. I'm unsure if this is intentional behaviour with how nodes are updated vs. created.`? Can we foresee instances when that would be an issue for users/cluster admins?

Also, for future reviewers, the [linked](https://github.com/kubernetes/enhancements/blob/master/keps/sig-windows/windows-runtimeclass-support.md#adding-new-label-nodekubernetesiowindows-build) section of the KEP describes the specifics of the desired label additions.

**Comment by gab-satchi** [comment]
> Can we foresee instances when that would be an issue for users/cluster admins?

When nodes are upgraded, they won't automatically pick up the new label. So if a user wants to use the new label as means for node selection, that won't work. Because it's just a label, mitigation would be to add the label manually.

**Comment by gab-satchi** [comment]
@PatrickLang User Story 1 for KEP 1302

**Comment by PatrickLang** [comment]
/assign @lavalamp @vishh
Can you approve for kubelet/api?

Thanks @gab-satchi

**Comment by josiahbjorgaard** [comment]
Since this is intended for v1.17, a quick reminder that code freeze is this Thursday at 5PM pacific time. Will this be merged by then?

**Comment by PatrickLang** [comment]
I pinged @lavalamp  and a few others in SIG-Node on slack. I also put it on SIG-Node agenda. Still trying to get an approver lined up.

File: pkg/kubelet/kubelet_node_status_windows.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/kubernetes/pkg/kubelet/winstats"
+)
+
+func addOSSpecificLabels(n *v1.Node) error {
+	osInfo, err := winstats.GetOSInfo()
+	if err != nil {
+		return err
+	}
+	n.Labels[v1.LabelWindowsBuild] = osInfo.GetBuild()
```
Comment (liggitt) [inline pkg/kubelet/kubelet_node_status_windows.go]: doesn't v1.LabelWindowsBuild need to be added to the list in updateDefaultLabels?

File: staging/src/k8s.io/api/core/v1/well_known_labels.go [hunk 2]
```diff
@@ -26,6 +26,8 @@ const (
 	LabelOSStable   = "kubernetes.io/os"
 	LabelArchStable = "kubernetes.io/arch"
 
+	LabelWindowsBuild = "node.kubernetes.io/windows-build"
```
Comment (liggitt) [inline staging/src/k8s.io/api/core/v1/well_known_labels.go]: add documentation of what this represents, the format, etc

File: pkg/kubelet/kubelet_node_status_windows.go [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+*/
+
+package kubelet
+
+import (
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/kubernetes/pkg/kubelet/winstats"
+)
+
+func addOSSpecificLabels(n *v1.Node) error {
```
Comment (liggitt) [inline pkg/kubelet/kubelet_node_status_windows.go]: suggest returning a map[string]string instead of taking a node object

**Comment by liggitt** [comment]
> While testing the functionality, I discovered that updating an existing node does not include the new label. Deleting the node and recreating does bring in the label. I'm unsure if this is intentional behaviour with how nodes are updated vs. created.

You need to add the new label key to the list of updated labels in the `updateDefaultLabels` function

**Comment by liggitt** [comment]
/approve

label looks good based on discussion in https://github.com/kubernetes/enhancements/pull/1302

/hold for comments
@PatrickLang has lgtm

**Comment by PatrickLang** [comment]
Thanks for updating based on the feedback!
/lgtm

**Pull request merged**